### PR TITLE
#65 with alternative event manager

### PR DESCRIPTION
**Pull request by wchill**
### Short Description:

Same as #65, but with an alternative event implementation.
- Added events
- Cherry-picked upstream
  - Item filter
- Separate logic for pokemon transfer
- Fixed walking #58

@OpenPoGo/maintainers

### CODE REVIEW

**Comment by meyer9** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/67/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by Reaver01** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/67/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

**Comment by jtdroste** [comment]
👍 

working!

[Approved with PullApprove](https://pullapprove.com/OpenPoGo/OpenPoGoBot/pull-request/67/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=OpenPoGo/OpenPoGoBot)

File: pokemongo_bot/plugins.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        plugins = self.get_available_plugins()
+        if plugin_name in plugins:
+            if plugin_name not in self.loaded_plugins:
+                module = imp.load_module(self.main_module, *plugins[plugin_name]['info'])
+                self.loaded_plugins[plugin_name] = {
+                    'name': plugin_name,
+                    'info': plugins[plugin_name]['info'],
+                    'module': module
+                }
+                self.logging.log('plugin "%s" loaded' % plugin_name)
```
Comment (jtdroste) [inline pokemongo_bot/plugins.py:58]: Should prefix this with "[x] "

**Pull request merged**